packages/compiler/server/serverlib.ts

```diff
@@ -766,14 +771,24 @@ export function createServer(host: ServerHost): Server {
     return references;
   }
 
-  function addKeywordCompletion(area: keyof KeywordArea, completions: CompletionList) {
+  function addKeywordCompletion(
+    area: keyof KeywordArea,
+    detail: string | undefined,
+    completions: CompletionList
+  ) {
     const filteredKeywords = keywords.filter(([_, x]) => area in x);
     for (const [keyword] of filteredKeywords) {
       completions.items.push({
         label: keyword,
         kind: CompletionItemKind.Keyword,
       });
     }
+    if (detail) {
+      completions.items.push({
```

Comment:
the detail needs to be added above when you are adding items, you are attaching the detail to each completion item, not adding a new completion item.